components/apimgt/org.wso2.carbon.apimgt.hostobjects/src/main/java/org/wso2/carbon/apimgt/hostobjects/APIStoreHostObject.java

```diff
@@ -4620,22 +4636,37 @@ public static NativeObject jsFunction_getDocument(Context cx, Scriptable thisObj
      * @param thisObj Scriptable object
      * @param args Passing arguments
      * @param funObj Function object
-     * @return String group id. 
-     * 
+     * @return NativeArray groupid array.
+     *
      */
-    public static String jsFunction_getGroupIds(Context cx, Scriptable thisObj, Object[] args, Function funObj) {
+    public static NativeArray jsFunction_getGroupIds(Context cx, Scriptable thisObj, Object[] args, Function funObj) {
         String response = (String) args[0];
         APIConsumer consumer = getAPIConsumer(thisObj);
-        String groupId = null;
+        NativeArray grpIdList = null;
+        String[] groupIdArray = null;
         try {
-            groupId = consumer.getGroupIds(response);
+            groupIdArray = consumer.getGroupIds(response);
+            if (groupIdArray != null) {
+                grpIdList = new NativeArray(0);
+                int i = 0;
+                for (String groupId : groupIdArray) {
+                    grpIdList.put(i, grpIdList, groupId);
+                    i++;
+                }
+                return grpIdList;
+            }else{
+                return null;
+            }
         } catch (APIManagementException e) {
-        	//This is actually not an exception, that should abort the user flow. If the groupId is not available then 
-        	//the flow for which the group id is not required will be run.
+            //This is actually not an exception, that should abort the user flow. If the groupId is not available then
+            //the flow for which the group id is not required will be run.
             log.error("Error occurred while getting group id", e);
```

Comment:
I think the comment is misleading here. I think the author is trying to say that this exception should not interrupt the flow and in case of an exception the flow which does not use group ID should work hence I will change the comment according to that and IMO it should be logged as a error